### PR TITLE
Setup dotnet-ci

### PR DESCRIPTION
**Pull request by natemcmaster**
Resolves https://github.com/aspnet/aspnet-docker/issues/355

AppVeyor doesn't support testing our 1709 images. Let's build PRs to this repo on dotnet-ci instead.

cc @mmitche / @pranavkm

### CODE REVIEW

File: build-pipeline/windows-pipeline.groovy [hunk 1]
```diff
@@ -0,0 +1,17 @@
+@Library('dotnet-ci') _
+
+// TODO: how to do this?
+//
+// if (params.RS3) {
+//   label('windows.10.amd64.serverrs3.open')
```
Comment (natemcmaster) [inline build-pipeline/windows-pipeline.groovy]: @mmitche / @MichaelSimons - what's the pipelines equivalent for this? https://github.com/dotnet/dotnet-docker/blob/master/netci.groovy#L34-L39

**Comment by natemcmaster** [comment]
cc @Eilon we'll likely need to enable some webhooks on this repo too.

**Comment by Eilon** [comment]
@natemcmaster - sure thing, let me know what to do and we can enable it. FYI I'm out on Friday, but will be back Monday.

File: build-pipeline/windows-pipeline.groovy (hunk 1 above)
Comment (mmitche) [reply build-pipeline/windows-pipeline.groovy]: simpleNode('windows.10.amd64.serverrs3.open') will do the same thing

File: build-pipeline/linux-pipeline.groovy [hunk 2]
```diff
@@ -0,0 +1,13 @@
+@Library('dotnet-ci') _
+
+simpleNode ('Ubuntu16.04', 'latest-or-auto-docker') {
+    stage ('Checking out source') {
+        checkout scm
+    }
+    stage ('Install pwsh') {
```
Comment (MichaelSimons) [inline build-pipeline/linux-pipeline.groovy]: This is modifying the state of the machine.  Should this get cleaned up at the end?  An alternative would be to utilize a Docker container.

Comment (mmitche) [reply build-pipeline/linux-pipeline.groovy]: Can it be installed to the local workspace directory?  That would also solve this issue.

Comment (natemcmaster) [reply build-pipeline/linux-pipeline.groovy]: Good point. I didn't know if the agents were going to sandbox test runs like Travis/AppVeyor do. I think it should be possible to download as a tarball as long as these dependencies are installed:

libc6, libgcc1, libgssapi-krb5-2, liblttng-ust0, libstdc++6, libcurl3, libunwind8, libuuid1, zlib1g, libssl1.0.0, libicu55

https://github.com/PowerShell/PowerShell/blob/master/docs/installation/linux.md#binary-archives

Comment (mmitche) [reply build-pipeline/linux-pipeline.groovy]: I think those should be available, if not we can get them.

**Comment by natemcmaster** [comment]
Updated

**Comment by pranavkm** [comment]
:shipit:

File: build-pipeline/linux-pipeline.groovy [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+simpleNode ('Ubuntu16.04', 'latest-or-auto-docker') {
+    stage ('Checking out source') {
+        checkout scm
+    }
+    stage ('Cleanup Docker') {
+        sh 'docker system prune -a -f'
+    }
+    stage ('Install pwsh locally') {
+        sh 'curl -fsSL https://github.com/PowerShell/PowerShell/releases/download/v6.0.1/powershell-6.0.1-linux-x64.tar.gz -o /tmp/powershell.tar.gz \\
```
Comment (mmitche) [inline build-pipeline/linux-pipeline.groovy]: Should triple quote this multi-line string

**Comment by mmitche** [comment]
@dotnet-bot test ci please

**Comment by mmitche** [comment]
Did you add the webhooks for this repo?

**Comment by Eilon** [comment]
I only started hooking up the CORS repo. Once we have that sorted I'll target more repos (and eventually all, presumably).

**Comment by Eilon** [comment]
@dotnet-bot test ci please

**Comment by natemcmaster** [comment]
@dotnet-bot test ci please

**Comment by natemcmaster** [comment]
@dotnet-bot test Windows 1709 2.0/nanoserver-1709 Build please

**Pull request merged**